tests/test_cli.py

```diff
@@ -192,6 +216,69 @@ def test_cli(self, case, lang):
                 case_output.unlink(missing_ok=True)
             exe.unlink(missing_ok=True)
 
+    @foreach(sorted(TEST_CASES))
+    # This test name must be alpha before `test_generated` otherwise
+    # KEEP_GENERATED does not work.
+    def test_env_cxx_gcc(self, case):
+        lang = "cpp"
+        ext = ".cpp"
+        if not os.path.exists(f"expected/{case}{ext}"):
+            raise unittest.SkipTest(f"expected/{case}{ext} not found")
+
+        env = os.environ.copy()
+        env["CXX"] = "g++-11" if sys.platform == "darwin" else "g++"
+        env["CXXFLAGS"] = "-std=c++14 -Wall -Werror"
+
+        if not spawn.find_executable(env["CXX"]):
+            raise unittest.SkipTest(f"{env['CXX']} not available")
+
+        settings = _get_all_settings(Mock(indent=4), env=env)[lang]
+        assert settings.linter[0].startswith("g++")
+
+        if not spawn.find_executable("astyle"):
+            raise unittest.SkipTest("astyle not available")
+
+        settings.formatter = ["astyle"]
```

Comment:
Looks like astyle is an alternative formatter and has nothing to do with whether CXX=gcc or clang.

Reply:
I'd like to remove the dependency on llvm, and clang-format is part of that goal. c.f. https://github.com/adsharma/py2many/issues/21  I guess adding a `--formatter=...` and `--linter=...` cli args is how we would expose that.  For now using astyle here ensures this group of tests doesnt use llvm.